raft/src/test/java/org/apache/kafka/snapshot/SnapshotsTest.java

```diff
@@ -75,4 +77,23 @@ public void testInvalidSnapshotFilenames() {
         // partition metadata
         assertEquals(Optional.empty(), Snapshots.parse(root.resolve("partition.metadata")));
     }
+
+    @Test
+    public void testDeleteSnapshot() throws IOException {
+
+        OffsetAndEpoch snapshotId = new OffsetAndEpoch(
+            TestUtils.RANDOM.nextInt(Integer.MAX_VALUE),
+            TestUtils.RANDOM.nextInt(Integer.MAX_VALUE)
+        );
+
+        Path logDirPath = TestUtils.tempDirectory().toPath();
+        FileRawSnapshotWriter snapshot = FileRawSnapshotWriter.create(logDirPath, snapshotId, Optional.empty());
+        snapshot.freeze();
```

Comment:
Let's close the snapshot writer. How about?

```suggestion
        try (FileRawSnapshotWriter snapshot = FileRawSnapshotWriter.create(logDirPath, snapshotId, Optional.empty())) {
            snapshot.freeze();
        }
```